### PR TITLE
Stabilize `const_float_methods`

### LINKED ISSUE #130843: Tracking Issue for `#![feature(const_float_methods)]`
Feature gate: `#![feature(const_float_methods)]`

This is a tracking issue for making `const` some methods of floating point types.

### Public API

```rust
// in `core`
impl <float> {
    pub const fn recip(self) -> Self;
    pub const fn to_degrees(self) -> Self;
    pub const fn to_radians(self) -> Self;
    pub const fn max(self, other: Self) -> Self;
    pub const fn min(self, other: Self) -> Self;
    pub const fn clamp(self, min: Self, max: Self) -> Self;
    pub const fn abs(self) -> Self;
    pub const fn signum(self) -> Self;
    pub const fn copysign(self, sign: Self) -> Self;
}
```

### Steps / History

- [x] Implementation: https://github.com/rust-lang/rust/pull/130568
- [ ] Final comment period (FCP)[^1]
- [ ] Stabilization PR

### Unresolved Questions

- None yet.

[^1]: https://std-dev-guide.rust-lang.org/feature-lifecycle/stabilization.html

### PR DESCRIPTION
**Pull request by eduardosm**
Tracking issue: https://github.com/rust-lang/rust/issues/130843
Relnotes: #133383

Stabilized const API:

```rust
// in `core`
impl f32/f64 {
    pub const fn recip(self) -> Self;
    pub const fn to_degrees(self) -> Self;
    pub const fn to_radians(self) -> Self;
    pub const fn max(self, other: Self) -> Self;
    pub const fn min(self, other: Self) -> Self;
    pub const fn clamp(self, min: Self, max: Self) -> Self;
    pub const fn abs(self) -> Self;
    pub const fn signum(self) -> Self;
    pub const fn copysign(self, sign: Self) -> Self;
}
```

Closes https://github.com/rust-lang/rust/issues/130843

r? libs-api

cc @RalfJung -- I think the way const-stability attributes work have change a bit since the last time a wrote a const-stabilization PR, please make sure I got them right.

### CODE REVIEW

File: library/core/src/intrinsics/mod.rs [hunk 1]
```diff
@@ -4267,7 +4267,11 @@ pub const fn minnumf16(_x: f16, _y: f16) -> f16 {
 /// The stabilized version of this intrinsic is
 /// [`f32::min`]
 #[rustc_nounwind]
-#[cfg_attr(bootstrap, rustc_const_unstable(feature = "const_float_methods", issue = "130843"))]
+#[cfg_attr(
+    bootstrap,
+    rustc_const_stable(feature = "const_float_methods", since = "CURRENT_RUSTC_VERSION")
+)]
+#[cfg_attr(not(bootstrap), rustc_intrinsic_const_stable_indirect)]
```
Comment (RalfJung) [inline library/core/src/intrinsics/mod.rs:4274]: These intrinsics have been approved by t-lang FCP [here](https://github.com/rust-lang/rust/issues/130843#issuecomment-2469761984).

File: library/core/src/num/f128.rs [hunk 2]
```diff
@@ -1292,7 +1287,6 @@ impl f128 {
     /// ```
     #[inline]
     #[unstable(feature = "f128", issue = "116909")]
```
Comment (RalfJung) [inline library/core/src/num/f128.rs:1289]:
```suggestion
    #[unstable(feature = "f128", issue = "116909")]
    #[rustc_const_unstable(feature = "f128", issue = "116909")]
```
Seems better just for consistency, and in case the intrinsic gets added.

`is_finite` will then also need the attribute.

File: library/core/src/num/f16.rs [hunk 3]
```diff
@@ -1269,7 +1264,6 @@ impl f16 {
     /// ```
     #[inline]
     #[unstable(feature = "f16", issue = "116909")]
```
Comment (RalfJung) [inline library/core/src/num/f16.rs:1266]: Same as the f128 case, seems better to mark this `rustc_const_unstable` so that it can use the intrinsic once that is reliable.

File: library/core/src/num/f32.rs [hunk 4]
```diff
@@ -879,8 +878,7 @@ impl f32 {
     /// ```
     #[must_use = "this returns the result of the comparison, without modifying either input"]
     #[stable(feature = "rust1", since = "1.0.0")]
-    #[rustc_const_unstable(feature = "const_float_methods", issue = "130843")]
-    #[inline]
```
Comment (RalfJung) [inline library/core/src/num/f32.rs:883]: Why did you remove the `inline`?

File: library/core/src/num/f16.rs [hunk 5]
```diff
@@ -326,7 +326,6 @@ impl f16 {
     #[inline]
     #[must_use]
     #[unstable(feature = "f16", issue = "116909")]
```
Comment (RalfJung) [inline library/core/src/num/f16.rs:328]:
```suggestion
    #[unstable(feature = "f16", issue = "116909")]
    #[rustc_const_unstable(feature = "f16", issue = "116909")]
```

**Review by RalfJung — COMMENTED** [review]
Cc @rust-lang/wg-const-eval

File: library/core/src/num/f32.rs (hunk 4 above)
Comment (eduardosm) [reply library/core/src/num/f32.rs:883]: Oops

**Comment by RalfJung** [comment]
Looks good, thanks. :)

@bors r+ rollup
(Given that this passed t-libs-api FCP, I don't think we need the reviewer to also be in t-libs-api.)
